nifi-nar-bundles/nifi-spark-bundle/nifi-livy-controller-service/src/main/java/org/apache/nifi/controller/livy/LivySessionController.java

```diff
@@ -389,9 +428,22 @@ private void manageSessions() throws InterruptedException, IOException {
                     log.debug("manageSessions() Registered new session: " + newSessionInfo);
                 }
             } else {
-                // Open one new session if there are no idle sessions
-                if (idleSessions == 0) {
-                    log.debug("manageSessions() There are " + numSessions + " sessions in the pool but none of them are idle sessions, creating...");
+                // If we exceeded our session pool size, look for `idle` sessions we can shut down
+                //  Two scenarios: we have no elastic pool sizing, in which case definitely look for candidates
+                //  Or, we do have elastic pool sizing, in which case we need to make sure we are above our max
+                //      pool size
+                if(idleSessions > 0 && numSessions > sessionPoolSize && (!elasticSessionPool || numSessions > maxSessionPoolSize)) {
```

Comment:
I'm not sure Is this conditional statement satisfies all possible cases...

e.g.) `idleSessions`=2, `numSessions`=6,  `elasticSessionPool`=true `maxSessionPoolSize`=4
  => we have to delete two idle sessions, but it only picks randomly one idle session and deletes it.

Reply:
It only sops 1 idle session at a time, that is true, but on the next run through, which happens every few seconds, it will stop the next one, and the next one, until it is back to where it is supposed to be.

Reply:
Thanks for your elaborate explanation.
I got your point.